asn1/InterledgerTypes.asn

```diff
@@ -0,0 +1,165 @@
+InterledgerTypes
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    Int16,
+    UInt8,
+    UInt64,
+    VarUInt,
+    Float64
+    FROM GenericTypes
+;
+
+-- Readable names for special characters that may appear in ILP addresses
+hyphen IA5String ::= "-"
+period IA5String ::= "."
+underscore IA5String ::= "_"
+tilde IA5String ::= "~"
+
+-- A standard interledger address
+Address ::= IA5String
+    (FROM
+        ( hyphen
+        | period
+        | "0".."9"
+        | "A".."Z"
+        | underscore
+        | "a".."z"
+        | tilde )
```

Comment:
They are ordered by value, i.e. their order in the ASCII table. E.g. `'A' == 0x41`, so it comes before `'_' == 0x5F`, which in turn comes before `'a' == 0x61`